### PR TITLE
fix(freeze): queue freeze/unfreeze does not refresh train

### PR DESCRIPTION
**Pull request by Syffe**
After freezing/unfreezing the queue, no events were sent in order to update the
summaries of pull requests. Thus, nothing was happening to the PRs until an unrelated
event was triggered to update their status.

Fixes MRGFY-1002

### CODE REVIEW

File: mergify_engine/queue/freeze.py [hunk 1]
```diff
@@ -131,10 +132,25 @@ async def save(self) -> None:
...
         )
+
+        await self.refresh_pulls(source="internal/queue_freeze_delete")
+
+        return result
+
+    async def refresh_pulls(self, source: str) -> None:
+
+        async for train in merge_train.Train.iter_trains(self.repository):
+            await super(type(train), train)._refresh_pulls(
```
Comment (Syffe) [inline mergify_engine/queue/freeze.py:153]:
```suggestion
            await super(type(train), train)._refresh_pulls()
```

I'm not sure accessing a protected method here is adequate, should I create a method that calls super() une `merge_train.Train`?

Comment (sileht) [reply mergify_engine/queue/freeze.py:153]: Just unprotect the current one.

File: mergify_engine/queue/freeze.py [hunk 2]
```diff
@@ -131,10 +132,25 @@ async def save(self) -> None:
...
         )
+
+        await self._refresh_pulls(source="internal/queue_freeze_delete")
+
+        return result
+
+    async def _refresh_pulls(self, source: str) -> None:
+
+        async for train in merge_train.Train.iter_trains(self.repository):
+            await super(type(train), train).refresh_pulls(
```
Comment (sileht) [inline mergify_engine/queue/freeze.py:153]:
```suggestion
            await train.refresh_pulls(
```

This should be enough, no ?

**Review by sileht — DISMISSED** [review]

File: mergify_engine/queue/freeze.py [hunk 3]
```diff
@@ -131,10 +132,25 @@ async def save(self) -> None:
...
+
+        await self._refresh_pulls(source="internal/queue_freeze_delete")
+
+        return result
+
+    async def _refresh_pulls(self, source: str) -> None:
+
+        async for train in merge_train.Train.iter_trains(self.repository):
+            await train.refresh_pulls(
+                repository=self.repository.repo,
```
Comment (sileht) [inline mergify_engine/queue/freeze.py:154]: Not related to your change, but I feel that repository here is useless as train already has the repository_ctxt. I will fix that.

**Review by sileht — APPROVED** [review]

**Comment by sileht** [comment]
@mergifyio rebase

**Comment by sileht** [comment]
@Mergifyio refresh

**Pull request merged**